js_modules/dagit/packages/core/src/partitions/SpanRepresentation.tsx

```diff
@@ -51,11 +52,15 @@ export function textToPartitions(selected: string, all: string[]) {
       result.push(term);
     }
   }
-  return result.sort((a, b) => all.indexOf(a) - all.indexOf(b));
+  return Array.from(new Set(result));
```

Comment:
Because I added a few optimizations based on selected.length === all.length, we need to make sure the selection doesn't have any duplicates. The order of the selected items never really mattered though as far as I can tell!